### PR TITLE
completed winding and core calculations

### PR DESCRIPTION
**Pull request by harshavardhanmantri**

### CODE REVIEW

File: frontend/backend/src/calculations.js [hunk 1]
```diff
@@ -2,7 +2,365 @@ import constants from "./constants"
...
+            model.output.winding.hv.wireInsulated2 = model.input.winding.hv.wireBare2 + model.input.winding.hv.insulation
+        }
+        return model
+    },
+    calculateCrossSection : function(model){
+        let x;
+        let y;
+        let g;
+        let h;
+        if(model.input.winding.lv.wireBare2 <= 16){
```
Comment (ganesh-dagadi) [inline frontend/backend/src/calculations.js]: not 16..it is 1.6

File: frontend/backend/src/calculations.js [hunk 2]
```diff
@@ -2,7 +2,365 @@ import constants from "./constants"
...
+        else {
+        if(model.input.winding.lv.insulationType === "FOIL"){
+            x = model.input.winding.lv.wireBare1*model.input.winding.lv.wireBare2
+        }
+        else{
+            x = model.input.winding.lv.wireBare1*model.input.winding.lv.wireBare2 - y
+        }
+
+        }
+        if(model.input.winding.hv.wireBare2 <= 16){
```
Comment (ganesh-dagadi) [inline frontend/backend/src/calculations.js]: not 16. It is 1.6

File: frontend/backend/src/calculations.js [hunk 3]
```diff
@@ -2,7 +2,365 @@ import constants from "./constants"
...
+        if(model.input.winding.hv.connectionType === "STAR"){
+            model.output.winding.hv.ratedCurrent = (model.input.general.power*1000)/(Math.sqrt(3)*model.input.winding.hv.voltage)
+        }
+        else{
+            model.output.winding.hv.ratedCurrent = (model.input.general.power*1000)/(3*model.input.winding.hv.voltage)
+        }
+        return model
+    },
+    calculateTurnPerLimb: function(model){
+        model.output.winding.lv.turnPerLimb = Math.round(model.input.winding.lv.calculateRatedCurrent/model.output.general.v_t)
```
Comment (ganesh-dagadi) [inline frontend/backend/src/calculations.js]: divide ratedvoltage/ v_t 
not rated current

File: frontend/backend/src/calculations.js [hunk 4]
```diff
@@ -2,7 +2,365 @@ import constants from "./constants"
...
+        else{
+            x = 0.97
+        }
+        model.output.stackingFactor = x
+        model.output.others.B_Six = Math.round((model.output.others.netCrossSection / model.output.others.stackingFactor)/
+        (model.input.coreWdg.w_d * 0.1) * 10)
+        return model
+    },
+    calculateI12 : function(model){
+        model.output.others.I12 = model.input.coreWdg.core2 + model.input.coreWdg.limbPlate1
```
Comment (ganesh-dagadi) [inline frontend/backend/src/calculations.js]: change core2 into w_d

File: frontend/backend/src/calculations.js [hunk 5]
```diff
@@ -2,7 +2,365 @@ import constants from "./constants"
...
+        model.output.winding.lv.strayLoss = (y ^ 4) * ((model.output.winding.lv.layers  *
+        model.input.winding.lv.noParallelRA1) ^ 2/9) * 100
+        let h = (model.input.winding.hv.wireBare2 / 10) * Math.sqrt(((model.input.winding.hv.wireBare1 *
+                (model.output.winding.hv.turnPerLayer + 1)) / model.output.winding.hv.windLength) *
+                model.output.winding.hv.layers * x)
+        model.output.winding.hv.strayLoss = (h ^ 4) * ((model.output.winding.hv.layers  *
+        model.input.winding.hv.noParallelRA1) ^ 2/9) * 100
+        return model
+    },
+    calculateConductorWeigth : function(model){
```
Comment (ganesh-dagadi) [inline frontend/backend/src/calculations.js]: change weight spelling

File: frontend/backend/src/calculations.js [hunk 6]
```diff
@@ -2,7 +2,365 @@ import constants from "./constants"
...
+        return model
+    },
+    calculateLoadLoss : function(model){
+        model.output.winding.lv.loadLoss = (model.output.winding.lv.ratedCurrent ^ 2) *
+        model.output.winding.lv.Resistance * 3 * (1 + model.output.winding.lv.strayLoss) 
+        model.output.winding.hv.loadLoss = (model.output.winding.hv.ratedCurrent ^ 2) *
+        model.output.winding.hv.Resistance * 3 * (1 + model.output.winding.hv.strayLoss) 
+        return model
+    },
+    calcuateS_am2 : function(model){
```
Comment (ganesh-dagadi) [inline frontend/backend/src/calculations.js]: change calculate spelling

File: frontend/backend/src/model.js [hunk 7]
```diff
@@ -125,6 +127,8 @@ const model = {
                 C_dist : 0.0,
                 Yoke_l : 0.0,
                 core1 : 0.0, //to be changed later
+                netCrossSection: 0.0,
+                stackingFactor : 0.0,
             }
         }
     }
```
Comment (ganesh-dagadi) [inline frontend/backend/src/model.js:138]: You cannot calculate exponent using ^ in javascript... u have to use Math.pow(equation , 2) for square. Please change that.
Also i just realised that cells J14 , J 16 ,J 18 and J 20 are not implemented. create new parameters in model.others called L_ID_D , L_OD_D  ,  H_ID_D , H_OD_D respectfully and write 4 more functions to calculate these..

**Review by ganesh-dagadi — CHANGES_REQUESTED** [review]
Please make the changes and push again

File: frontend/backend/src/calculations.js (hunk 1 above)
Comment (harshavardhanmantri) [reply frontend/backend/src/calculations.js]: done

File: frontend/backend/src/calculations.js (hunk 2 above)
Comment (harshavardhanmantri) [reply frontend/backend/src/calculations.js]: done

File: frontend/backend/src/calculations.js (hunk 3 above)
Comment (harshavardhanmantri) [reply frontend/backend/src/calculations.js]: done

File: frontend/backend/src/calculations.js (hunk 4 above)
Comment (harshavardhanmantri) [reply frontend/backend/src/calculations.js]: done

File: frontend/backend/src/calculations.js (hunk 5 above)
Comment (harshavardhanmantri) [reply frontend/backend/src/calculations.js]: done

File: frontend/backend/src/calculations.js (hunk 6 above)
Comment (harshavardhanmantri) [reply frontend/backend/src/calculations.js]: done

File: frontend/backend/src/model.js (hunk 7 above)
Comment (harshavardhanmantri) [reply frontend/backend/src/model.js:138]: done

**Review by ganesh-dagadi — APPROVED** [review]

**Pull request merged**